src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
In use this will look like `indexPatterns.getSavedIndexPatterns()`. I don't think that communicates that the results of that call will not be instances of `IndexPattern`, but will be saved objects, with `id`, `type`, `attributes`, etc. Perhaps that is another source of confusion here, the `IndexPatterns` service normally gives us `IndexPattern` objects, not `SavedObjects`... 

Maybe we should just be using the `savedObjectsClient` for all of these rather than force them into the `indexPatterns` service...